### PR TITLE
add: auth features to the social media app

### PR DESCRIPTION
**Pull request by santhoshpemmaka**
Hi folks. I implemented login and signup features to the social media app.

Files Check: [Review these files only](https://github.com/santhoshpemmaka/NamasteSocialApp/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.jsx&show-viewed-files=true)

image

image

### CODE REVIEW

**Review by amartya-rs — APPROVED** [review]
Great job, Santosh. Code looks absolutely fine to me.
Good to merge 👍

File: src/utils/server-request.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+export const loginUserRequest = (email, password) => {
+	return axios.post("api/auth/login", {
+		username: email,
+		password: password,
+	});
+};
+
+export const signUpRequest = (email, password, firstName, lastName) => {
+	return axios.post("api/auth/signup", {
```
Comment (VLeads) [inline src/utils/server-request.js:11]: It's great that you have placed all apis at one place.

File: src/Features/Auth.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+const authSlice = createSlice({
+	name: "social-app-auth",
+	initialState,
+	reducers: {},
+	extraReducers: {
+		[loginUser.pending]: (state) => {
+			state.authStatus = "pending";
+		},
+		[loginUser.fulfilled]: (state, action) => {
+			state.authStatus = "fulfilled";
```
Comment (VLeads) [inline src/Features/Auth.js:61]: Appreciate: You are covering all states "pending, fulfilled, rejected"

**Review by VLeads — APPROVED** [review]
Overall code is clean. You can merge it !!

**Pull request merged**